### ISSUE TITLE
Commissions Export/Reporting Functionality

### ISSUE THREAD

**Issue by davidsherlock** [easydigitaldownloads/EDD-Commissions#174]
Within EDD Reports screen it would be useful to have the ability to export all the commissions records for a given date range and status (paid, unpaid and revoked), should the commissions plugin be installed.

The export would effectively contain what is currently shown on the commissions list within the admin panel.

**Comment by ghost1227** [comment]
Note: Sum totals, not displaying actual commissions

**Comment by ghost1227** [comment]
Month to month if selected range is greater than a month, day to day otherwise.

**Comment by ghost1227** [comment]
SEE: Earnings export in 2.7

**Comment by ghost1227** [comment]
@pippinsplugins First pass, I THINK that does what you want. I should note that as is, it doesn't actually work. @cklosowski recently adjusted Commissions to load with a priority of `11` on `plugins_loaded`. This results in it not triggering the `edd_register_batch_exporter` hook correctly. If you drop Commissions to load on `10`, it works fine. Not sure what the right solution for this is... thoughts?

**Comment by pippinsplugins** [comment]
@ghost1227 Looking good. The counts are all look right but we need to add value as well. So fore each month and each status, we need to have a count and a summed value column.

**Comment by davidsherlock** [comment]
@pippinsplugins, what format are the exports going to be? Is it just going to he summed totals or itemised (as per the commission admin screen)? The later would be the most useful for me. Specifically if filtering was available such as export by vendor, status, date range, etc.

**Comment by pippinsplugins** [comment]
It'll be summed amounts + counts for each month.

Itemized exports are already supported and have been for a long time :)

**Comment by davidsherlock** [comment]
I mean itemized exports of the commission records, I can't see anywhere to achieve this - maybe I am missing something? :)

**Comment by pippinsplugins** [comment]
@davidsherlock I'm not sure what you mean. Could you elaborate?

**Comment by davidsherlock** [comment]
Sure, within the commissions panel (admin panel), you can see the following data presented in the table:

- User
- Download
- Payment
- Rate
- Amount	
- Status
- Date

However the commissions screen only shows 10 items per page and there is no functionality to show 100 items for example. 

It would be really useful to have be able to export full itemized CSV files for accounting and archiving purposes containing this data. 

I feel this should be added to the EDD > Reports > Export if the commissions plugin is installed, with the ability to set a date range, commission status (all, unpaid, paid or revoked) and 'potentially' a vendor select.

**Comment by ghost1227** [comment]
@pippinsplugins added amounts

**Comment by pippinsplugins** [comment]
Looks great! Just two more change requests:

1. Change `Net Amount` to `Net Total`
2. Add a `Gross Total` column before `Net Total`

**Comment by pippinsplugins** [comment]
@davidsherlock thank you for the feedback! We'll need to do that as a separate issue, perhaps in 3.4.

**Comment by ghost1227** [comment]
@pippinsplugins Ready for testing again!

**Comment by pippinsplugins** [comment]
Great! Just one item left. Now there is an empty column at the end of the CSV:
<img width="928" alt="screen shot 2017-02-20 at 10 00 32 am" src="https://cloud.githubusercontent.com/assets/1034109/23132520/79cba532-f753-11e6-83cd-218393c861d3.png">

**Comment by ghost1227** [comment]
@pippinsplugins Good catch! Fixed!

**Comment by pippinsplugins** [comment]
Perfect, merged!.

**Comment by ghost1227** [comment]
Awesome!

**Comment by cklosowski** [comment]
Re-opening just to get some testing done once #178 is done.

**Comment by SDavisMedia** [comment]
Correct me if I'm wrong here. Testing on `issue/178`, the Paid and Unpaid data looks backwards. 

One unpaid commission record and its export:
<img width="1219" alt="screenshot 2017-02-21 08 11 59" src="https://cloud.githubusercontent.com/assets/2359131/23173261/716357ea-f80d-11e6-9d7e-52994c28fd2c.png">
<img width="786" alt="screenshot 2017-02-21 08 12 42" src="https://cloud.githubusercontent.com/assets/2359131/23173281/8e453450-f80d-11e6-8287-0111f2233ddc.png">

Now I'm going to mark that single record as paid and export again:
<img width="1217" alt="screenshot 2017-02-21 08 13 36" src="https://cloud.githubusercontent.com/assets/2359131/23173316/a83a9062-f80d-11e6-934c-4cee4f87c697.png">
<img width="783" alt="screenshot 2017-02-21 08 14 08" src="https://cloud.githubusercontent.com/assets/2359131/23173347/bd4600d6-f80d-11e6-97ed-013ebcd440dc.png">

**Comment by SDavisMedia** [comment]
Revoked looks fine.

**Comment by pippinsplugins** [comment]
Definitely backwards @ghost1227

**Comment by ghost1227** [comment]
Good catch, fixed in #178

**Comment by SDavisMedia** [comment]
All good on all fronts. :+1:

**Comment by pippinsplugins** [comment]
Something isn't quite working here.

I did an export that starts in `Jan 2016` and ends in `Feb 2017`. All months in 2016 show `0` for the gross paid amounts, even though those months are definitely not 0.

<img width="810" alt="screen shot 2017-02-22 at 11 06 55 am" src="https://cloud.githubusercontent.com/assets/1034109/23222917/0c79c962-f8ef-11e6-897a-670a92992eb7.png">

**Comment by pippinsplugins** [comment]
Export `Jan 2016` to `Dec 2016` gives the same results, so it's not a year boundary issue.

**Comment by ghost1227** [comment]
@pippinsplugins fixed!

**Comment by pippinsplugins** [comment]
Testing!

**Comment by pippinsplugins** [comment]
Perfect!
